pkg/compactor/block_upload_test.go

```diff
@@ -0,0 +1,669 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package compactor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/gorilla/mux"
+	"github.com/oklog/ulid"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/thanos-io/thanos/pkg/block"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/weaveworks/common/user"
+
+	"github.com/grafana/mimir/pkg/storage/bucket"
+	mimir_tsdb "github.com/grafana/mimir/pkg/storage/tsdb"
+)
+
+// Test MultitenantCompactor.HandleBlockUpload with uploadComplete=false (the default).
+func TestMultitenantCompactor_HandleBlockUpload_Create(t *testing.T) {
+	const tenantID = "test"
+	const blockID = "01G3FZ0JWJYJC0ZM6Y9778P6KD"
+	bULID := ulid.MustParse(blockID)
+	now := time.Now().UTC().UnixMilli()
```

Comment:
Nit: no need to call UTC() for conversion to unix timestamp.

Reply:
@pstibrany oops, thanks!

Reply:
Done.